types/testing.go

```diff
@@ -67,6 +66,37 @@ func NewMockSignersAndKeyInfo(numSigners int) (MockSigner, []KeyInfo) {
 	return signer, ki
 }
 
+// MustGenerateMixedKeyInfo produces m bls keys and n secp keys.
+// BLS and Secp will be interleaved. The keys will be valid, but not deterministic.
```

Comment:
Non-determinism is really sucky for tests (e.g. block CIDs can't be relied upon). Please make this deterministic or file and link to an issue for whatever makes that currently impossible.

Reply:
Filed #3570 for this.  Easy way is to generate in advance and store in a package constant.  Hard way is to change BLS sigs interface and write some rust.

Reply:
Thanks @ZenGround0!